python/create_service_diagnostics_bundle.sh

```diff
@@ -1,26 +1,73 @@
 #!/usr/bin/env bash
 
-set -eu -o pipefail
+# Script that gathers SDK-related service diagnostics artifacts.
 
-readonly SCRIPT_DIRECTORY="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null && pwd)"
-readonly DCOS_SERVICE_DIAGNOSTICS_SCRIPT_PATH="dcos-sdk-service-diagnostics/python"
+# Exit immediately if a command exits with a non-zero status, or zero if all
+# commands in the pipeline exit successfully.
+set -eu -o pipefail
 
+# Script version goes here.
 readonly VERSION='v0.6.0'
 
-readonly BUNDLES_DIRECTORY="service-diagnostic-bundles"
-readonly DOCKER_IMAGE="mesosphere/dcos-sdk-service-diagnostics:${VERSION}"
-readonly SCRIPT_NAME="create_service_diagnostics_bundle.py"
-readonly HOST_DCOS_CLI_DIRECTORY="${DCOS_DIR:-${HOME}/.dcos}"
-readonly TTY_OPTS="${TTY_OPTS:=-it}"
+# ###
+# 1. section: Check requirements before main execution.
+# ###
+readonly REQUIREMENTS=("docker" "dcos")
 
-readonly CONTAINER_BUNDLES_DIRECTORY="/${BUNDLES_DIRECTORY}"
-readonly CONTAINER_DCOS_CLI_DIRECTORY_RO="/dcos-cli-directory"
-readonly CONTAINER_DCOS_CLI_DIRECTORY="/root/.dcos"
+for requirement in ${REQUIREMENTS[*]}; do
+  if ! [[ -x $(command -v "${requirement}") ]]; then
+    echo "You need to install '${requirement}' to run this script"
+  fi
+done
+
+# ###
+# 2. section: Define readonly variables (script configuration).
+# ###
+readonly SCRIPT_DIRECTORY="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null && pwd)"
+readonly DCOS_SERVICE_DIAGNOSTICS_SCRIPT_PATH="dcos-sdk-service-diagnostics/python"
+readonly BUNDLES_DIRECTORY="service-diagnostic-bundles"
+readonly PYTHON_SCRIPT_NAME="create_service_diagnostics_bundle.py"
 
+# ###
+# 3. section: Define functions.
+# ###
 function is_development_mode() {
   [[ ${SCRIPT_DIRECTORY} = *${DCOS_SERVICE_DIAGNOSTICS_SCRIPT_PATH} ]]
 }
 
+function run_in_container() {
+  local TTY_OPTS="${TTY_OPTS:=-it}"
+  local DOCKER_IMAGE="mesosphere/dcos-sdk-service-diagnostics:${VERSION}"
+
+  # Docker volumes configuration
+  local HOST_DCOS_CLI_DIRECTORY="${DCOS_DIR:-${HOME}/.dcos}"
+  local CONTAINER_BUNDLES_DIRECTORY="/${BUNDLES_DIRECTORY}"
+  local CONTAINER_DCOS_CLI_DIRECTORY_RO="/dcos-cli-directory"
+
+  docker run \
+          "${TTY_OPTS}" \
+          --rm \
+          -v "$(pwd)/${BUNDLES_DIRECTORY}:${CONTAINER_BUNDLES_DIRECTORY}" \
+          -v "${HOST_DCOS_CLI_DIRECTORY}:${CONTAINER_DCOS_CLI_DIRECTORY_RO}":ro \
+          ${CONTAINER_DCOS_SERVICE_DIAGNOSTIC_VOLUME_MOUNT} \
+          -e PYTHONPATH=${CONTAINER_DCOS_SERVICE_DIAGNOSTICS_DIRECTORY}/dcos-commons/testing:${CONTAINER_DCOS_SERVICE_DIAGNOSTICS_DIRECTORY} \
+          "${DOCKER_IMAGE}" \
+          sh -l -c "${*:-}"
+}
+
+function show_version() {
+  # Print script version.
```

Comment:
I'm not a great fan of comments that state the obvious :-/

Reply:
agreed